walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -21,69 +21,113 @@
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    List<TimeRange> possibleTimes = new ArrayList<>();
-    List<TimeRange> conflictingTimes = new ArrayList<>();
-    Collection<String> attendees = request.getAttendees();
+    List<TimeRange> conflictingTimes = findConflictingTimes(events, request.getAttendees());
+    List<TimeRange> possibleTimes = findPossibleTimes(conflictingTimes, request.getDuration());
+
+    return possibleTimes;
+  }
+
+  public List<TimeRange> findConflictingTimes(Collection<Event> events,
```

Comment:
For method return types, you should generally use the immutable type (such as ImmutableList) instead of the general collection interface type (such as List). This communicates to your callers the semantic guarantees of immutable types, which is almost always very useful information. When it's a general collection type, callers may be under the impression that they can mutate the return value, leading to failures at runtime.